### PR TITLE
Check tail calls

### PR DESCRIPTION
**Pull request by elazarg**
Check the caller part  of #117.
This PR adds a dedicated type for FDs of maps-to-programs, and checks them.

Signed-off-by: Elazar Gershuni <elazarg@gmail.com>

### CODE REVIEW

**Comment by elazarg** [comment]
The new type is not strictly necessary, but I think the types are conceptually very different, and are never used in similar contexts.

File: src/crab_utils/debug.cpp [hunk 1]
```diff
@@ -8,6 +8,6 @@ std::set<std::string> CrabLog;
 
 unsigned CrabVerbosity = 0;
 
-bool CrabWarningFlag = true;
+bool CrabWarningFlag = false;
```
Comment (dthaler) [inline src/crab_utils/debug.cpp:11]: why this change?

**Review by dthaler — APPROVED** [review]

File: src/crab_utils/debug.cpp (hunk 1 above)
Comment (elazarg) [reply src/crab_utils/debug.cpp:11]: I find these warnings not useful most of the time - they mostly warn about values becoming TOP, which does not hurt soundness, and it clutters the output of the tests. 
(BTW I've just pushed this change in #245. Of course it can be reverted).

**Comment by dthaler** [comment]
@elazarg are there test cases for this functionality?

**Comment by elazarg** [comment]
No dedicated ones, only the programs from the benchmarks (and all are negative - no expected failures). Those that I looked at seemed fine:
```
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/1
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/3
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/4
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/5
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/6
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/7
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o 2/10
ebpf-samples/bpf_cilium_test/bpf_lxc_jit.o from-container
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/2
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/3
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/4
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/5
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/6
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o 2/7
ebpf-samples/bpf_cilium_test/bpf_lxc-DUNKNOWN.o from-container
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/2
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/3
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/4
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/5
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/6
ebpf-samples/bpf_cilium_test/bpf_lxc-DDROP_ALL.o 2/7
ebpf-samples/bpf_cilium_test/bpf_netdev.o 2/2
ebpf-samples/bpf_cilium_test/bpf_netdev.o 2/3
ebpf-samples/bpf_cilium_test/bpf_netdev.o 2/4
ebpf-samples/bpf_cilium_test/bpf_netdev.o 2/5
ebpf-samples/bpf_cilium_test/bpf_netdev.o 2/7
ebpf-samples/bpf_cilium_test/bpf_overlay.o 2/2
ebpf-samples/bpf_cilium_test/bpf_overlay.o 2/3
ebpf-samples/bpf_cilium_test/bpf_overlay.o 2/4
ebpf-samples/bpf_cilium_test/bpf_overlay.o 2/5
ebpf-samples/bpf_cilium_test/bpf_overlay.o 3/2
ebpf-samples/bpf_cilium_test/bpf_lb-DLB_L3.o 2/2
ebpf-samples/bpf_cilium_test/bpf_lb-DLB_L4.o 2/2
ebpf-samples/bpf_cilium_test/bpf_lb-DUNKNOWN.o 2/2
ebpf-samples/cilium/bpf_lb.o 2/1
ebpf-samples/cilium/bpf_lb.o from-netdev
ebpf-samples/cilium/bpf_lxc.o 2/1
ebpf-samples/cilium/bpf_lxc.o 2/3
ebpf-samples/cilium/bpf_lxc.o 2/4
ebpf-samples/cilium/bpf_lxc.o 2/5
ebpf-samples/cilium/bpf_lxc.o 2/6
ebpf-samples/cilium/bpf_lxc.o 2/7
ebpf-samples/cilium/bpf_lxc.o 2/8
ebpf-samples/cilium/bpf_lxc.o 2/9
ebpf-samples/cilium/bpf_lxc.o 2/10
ebpf-samples/cilium/bpf_lxc.o 2/11
ebpf-samples/cilium/bpf_lxc.o 2/12
ebpf-samples/cilium/bpf_netdev.o 2/1
ebpf-samples/cilium/bpf_netdev.o 2/3
ebpf-samples/cilium/bpf_netdev.o 2/4
ebpf-samples/cilium/bpf_netdev.o 2/5
ebpf-samples/cilium/bpf_netdev.o 2/7
ebpf-samples/cilium/bpf_overlay.o 2/1
ebpf-samples/cilium/bpf_overlay.o 2/3
ebpf-samples/cilium/bpf_overlay.o 2/4
ebpf-samples/cilium/bpf_overlay.o 2/5
ebpf-samples/cilium/bpf_overlay.o 2/7
ebpf-samples/linux/sockex2_kern.o socket2
ebpf-samples/linux/sockex3_kern.o socket/3
ebpf-samples/linux/sockex3_kern.o socket/4
ebpf-samples/linux/sockex3_kern.o socket/1
ebpf-samples/linux/sockex3_kern.o socket/2
ebpf-samples/linux/tracex4_kern.o kretprobe/kmem_cache_alloc_node
ebpf-samples/ovs/datapath.o tail-0
ebpf-samples/ovs/datapath.o tail-1
ebpf-samples/ovs/datapath.o tail-2
ebpf-samples/ovs/datapath.o tail-3
ebpf-samples/ovs/datapath.o tail-4
ebpf-samples/ovs/datapath.o tail-5
ebpf-samples/ovs/datapath.o tail-7
ebpf-samples/ovs/datapath.o tail-8
ebpf-samples/ovs/datapath.o tail-12
ebpf-samples/ovs/datapath.o tail-13
ebpf-samples/ovs/datapath.o tail-32
ebpf-samples/ovs/datapath.o af_xdp
ebpf-samples/ovs/datapath.o downcall
ebpf-samples/ovs/datapath.o egress
ebpf-samples/suricata/xdp_filter.o xdp
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_accept4_e
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_empty
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_pread64_e
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_preadv64_e
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_pwrite64_e
ebpf-samples/falco/probe.o raw_tracepoint/filler/sys_single_x
ebpf-samples/falco/probe.o raw_tracepoint/filler/terminate_filler
ebpf-samples/falco/probe.o raw_tracepoint/page_fault_kernel
ebpf-samples/falco/probe.o raw_tracepoint/page_fault_user
ebpf-samples/falco/probe.o raw_tracepoint/sched_switch
```

**Comment by dthaler** [comment]
I'm adding a negative test case, will have a PR today
